src/grocerease.cpp

```diff
@@ -0,0 +1,330 @@
+/*
+ * Copyright (C) 2022 Carry Assistant Trolley (C.A.T) Technologies
+ *  
+ * Jia Wei Tan, February 16, 2022.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02111, USA.
+ */
+
+/*General Purpose Input Outsput for Raspberry Pi */
+
+
+#include "cart.h"
+#include "grocerease.h"
+#include <iostream>
+#include <chrono>
+#include <cstdlib>
+Cart cart;
+
+void Grocerease::initialize()
+{
+    cart.start();
+    read_RSSI();
+    cart.compute_angle(a,b,c);
+    angle_new = cart.getAngle();
+    angle_robot = angle_new;
+    angle_old = angle_new;
+}
+
+void Grocerease::initialize_test(float angle_feed, float angle_robot_feed, float dist_a_feed)
+{
+    cart.start();
+    angle_new = angle_feed;
+    angle_robot = angle_robot_feed;
+    angle_old = angle_new;
+    angle_diff = angle_new - angle_robot;
+}
+
+void Grocerease::run_program()
+{ 
+    read_RSSI();
+    cart.compute_angle(a,b,c);
+    write_newAngle();
+    write_a_distance();
+
+    if (a_distance <= followDistance)
+    {
+        amendAngle();
+        readCurrentRotation();
+        angleRecompute();
+        angleDifference();
+        rotateDecision();
+        rotation();
+    }
+    else
+    {
+        amendTime();
+        readIR();
+        ObstacleAvoidance();
+    }
+    angle_old = angle_new;  
+}
+
```

Comment:
@addy1997 @AzvenusK I feel run_program() and run_program_test() might not compile directly like this.
@jwtan5 Can you check if grocerase.run_program works in your main.cpp() ??

I feel that i might be better to have two files like RSSI_IR.cpp and test__RSSI_IR.cpp where we write lines 52-72 there directly by calling objects of grocerease class.?



What do you think? If this works, then no need to change. But if you all feel it might be wrong, then we maybe can use this^^^

Let me know ASAP

Reply:
@sarthakahuja11 runs perfectly on the pi